spyder/app/tour.py

```diff
@@ -982,13 +1044,22 @@ def _process_widgets(self, names, spy_window):
         dockwidgets = []
 
         for name in names:
-            base = name.split('.')[0]
-            temp = getattr(spy_window, base)
-
-            # Check if it is the current editor
-            if 'get_current_editor()' in name:
-                temp = temp.get_current_editor()
-                temp = getattr(temp, name.split('.')[-1])
+            try:
+                base = name.split('.')[0]
+                try:
+                    temp = getattr(spy_window, base)

```

Comment:
```suggestion
                    temp = getattr(spy_window, name)
```

So that execution passes properly to the outer `except` block if `name` isn't a simple attribute (it only happened to work as is because `editor` happened to be part of the particular string we passed).